main.go

```diff
@@ -135,78 +224,76 @@ func wrapMain() error {
+	err = runner.Run(gcloudCmd, "container", "clusters", "get-credentials", c.String("cluster"), "--project", project, "--zone", c.String("zone"))
 	if err != nil {
 		return fmt.Errorf("Error: %s\n", err)
 	}
 
 	data := map[string]interface{}{
-		// http://readme.drone.io/usage/variables/#string-interpolation:2b8b8ac4006be88c769f5e3fd99b009a
-		"BUILD_NUMBER": build.Number,
-		"COMMIT":       build.Commit,
-		"BRANCH":       build.Branch,
-		"TAG":          "", // How?
+		"BUILD_NUMBER": c.String("drone-build-number"),
+		"COMMIT":       c.String("drone-commit"),
+		"BRANCH":       c.String("drone-branch"),
+		"TAG":          c.String("drone-tag"),
 
 		// https://godoc.org/github.com/drone/drone-plugin-go/plugin#Workspace
-		"workspace": workspace,
-		"repo":      repo,
-		"build":     build,
-		"system":    system,
+		// TODO do we really need these?
+		// "workspace": workspace,
+		// "repo":      repo,
+		// "build":     build,
+		// "system":    system,
 
 		// Misc useful stuff.
-		// Note that we don't include all of the vargs, since that includes the GCP token.
-		"project":   vargs.Project,
-		"zone":      vargs.Zone,
-		"cluster":   vargs.Cluster,
-		"namespace": vargs.Namespace,
+		// note that secrets like gcp token are excluded
+		"project":   project,
+		"zone":      c.String("zone"),
+		"cluster":   c.String("cluster"),
+		"namespace": c.String("namespace"),
 	}
 
-	for k, v := range vargs.Vars {
+	secretsAndData := map[string]interface{}{}
+	for k, v := range vars {
 		// Don't allow vars to be overridden.
 		// We do this to ensure that the built-in template vars (above) can be relied upon.
 		if _, ok := data[k]; ok {
 			return fmt.Errorf("Error: var %q shadows existing var\n", k)
 		}
 
 		data[k] = v
+		secretsAndData[k] = v
 	}
 
-	if vargs.Verbose {
+	if c.Bool("verbose") {
 		dump := data
 		delete(dump, "workspace")
 		dumpData(os.Stdout, "DATA (Workspace Values Omitted)", dump)
 	}
 
-	secrets := map[string]interface{}{}
-	for k, v := range vargs.Secrets {
+	for k, v := range secrets {
+		if _, ok := secretsAndData[k]; ok {
+			return fmt.Errorf("Error: secret var %q shadows existing var\n", k)
+		}
 		if v == "" {
 			return fmt.Errorf("Error: secret var %q is an empty string\n", k)
 		}
 
 		// Base64 encode secret strings.
-		secrets[k] = base64.StdEncoding.EncodeToString([]byte(v))
-	}
-	for k, v := range vargs.SecretsBase64 {
-		if _, ok := secrets[k]; ok {
-			return fmt.Errorf("Error: secret var %q is already set in Secrets\n", k)
-		}
-		if v == "" {
-			return fmt.Errorf("Error: secret var %q is an empty string\n", k)
-		}
-		// Don't base64 encode these secrets, they already are.
-		secrets[k] = v
+		secretsAndData[k] = v
 	}
 
 	mapping := map[string]map[string]interface{}{
-		vargs.Template:       data,
-		vargs.SecretTemplate: secrets,
+		kubeTemplate:   data,
+		secretTemplate: secretsAndData,
```

Comment:
This is a significant difference in behavior with the old version. Previously, only secrets were passed to secret templates, now secret templates have access to both variables and secrets.